src/guest_memory.rs

```diff
@@ -175,6 +178,11 @@ pub trait GuestMemoryRegion: Bytes<MemoryRegionAddress, E = Error> {
     unsafe fn as_mut_slice(&self) -> Option<&mut [u8]> {
         None
     }
+
+    /// Convert an absolute address in an address space (GuestMemory) to a host pointer,
+    /// or return None if it is out of bounds. It always returns None when `DIRECT_ACCESS` is false.
+    /// The returned pointer may be used for ioctls etc.
+    fn get_host_address(&self, addr: MemoryRegionAddress) -> Option<*mut u8>;
```

Comment:
This can be implemented in terms of `as_mut_slice()`. Also please make the result `Option<NonNull<u8>>`.